recipes/osgearth/all/conanfile.py

```diff
@@ -23,192 +25,166 @@ class OsgearthConan(ConanFile):
+        export_conandata_patches(self)
 
-    @property
-    def _source_subfolder(self):
-        return "source_subfolder"
+    def validate(self):
+        check_min_cppstd(self, 14)
 
-    @property
-    def _build_subfolder(self):
-        return "build_subfolder"
+        # The official Conan recipe for rocksdb will delete all of the C++ headers from the packaged include directory
+        # in a shared build. osgearth assumes that these headers are available, however. What this means is that we
+        # cannot currently support shared builds of rocksdb.
+        if self.options.build_rocksdb_cache:
+            is_rocksdb_shared = self.dependencies["rocksdb"].options.shared
 
-    def validate(self):
-        if self.settings.compiler.get_safe("cppstd"):
-            tools.check_min_cppstd(self, 11)
-        elif self.settings.compiler == "apple-clang":
-            raise ConanInvalidConfiguration("With apple-clang cppstd needs to be set, since default is not at least c++11.")
+            if is_rocksdb_shared:
+                raise ConanInvalidConfiguration('The "build_rocksdb_cache" option cannot be used with a shared build of rocksdb.')
 
     def configure(self):
         if self.options.shared:
-            del self.options.fPIC
+            self.options.rm_safe("fPIC")
 
     def config_options(self):
         if self.settings.os != "Windows":
             self.options.enable_wininet_for_http = False
 
         if self.settings.os == "Windows":
-            del self.options.fPIC
-
-        if self.settings.compiler == "Visual Studio":
-            self.options.build_procedural_nodekit = False
-
-        if self.settings.compiler == "gcc" and self.settings.compiler.version == "11":
-            # need draco >= 1.4.0 for gcc11
-            # https://github.com/google/draco/issues/635
-            self.options.with_draco = False
+            self.options.rm_safe("fPIC")
 
     def requirements(self):
-
         self.requires("opengl/system")
-        self.requires("gdal/3.4.3")
-        self.requires("openscenegraph/3.6.5")
-        self.requires("libcurl/8.2.0")
-        self.requires("lerc/2.2")
+        self.requires("gdal/3.10.3")
+        self.requires("openscenegraph/3.6.5", transitive_headers=True)
+        self.requires("libcurl/[>=7.78.0 <9]")
+        self.requires("lerc/4.0.1")
         self.requires("rapidjson/1.1.0")
+        self.requires("sqlite3/[>=3.42 <4]")
 
-        self.requires("zlib/[>=1.2.11 <2]")  # override
-        self.requires("libtiff/4.5.1")  # override
-        self.requires("libpng/1.6.40")  # override
-
-        # if self.options.build_triton_nodekit:
-        #     self.requires("triton_nodekit")
-        # if self.options.build_silverlining_nodekit:
-        #     self.requires("silverlining_nodekit")
-        if self.options.build_leveldb_cache:
-            self.requires("leveldb/1.22")
         if self.options.build_rocksdb_cache:
             self.requires("rocksdb/6.29.5")
```

Comment:
```suggestion
            self.requires("rocksdb/10.5.1")
```

It would be great checking it if work with RockSDB 10.5.1 due the fix for headers